File: src/ts/numberOfSteps.ts
export function numberOfSteps (num: number): number {
  let result = 0
  while (num !== 0) {
    result += ((num & 1) === 1) ? 2 : 1
    num >>>= 1
  }

  return Math.max(result - 1, 0)
}

export function numberOfSteps1 (num: number): number {
  let result = -1
  while (num !== 0) {
    result += ((num & 1) === 1) ? 2 : 1
    num >>>= 1
  }

  return Math.max(result, 0)
}

File: src/rs/number_of_steps.rs
#[allow(dead_code)]
pub struct NumberOfSteps {}

#[allow(dead_code)]
impl NumberOfSteps {
    pub fn number_of_steps(num: i32) -> i32 {
        let mut n = num as u32;
        let mut result = -1;

        while n > 0 {
            result += if (n & 1) == 1 { 2 } else { 1 };

            n >>= 1;
        }

        result.max(0)
    }
}
